lib/fennec/evaluator/allocate/request.ex

```diff
@@ -0,0 +1,112 @@
+defmodule Fennec.Evaluator.Allocate.Request do
+  @moduledoc false
+
+  alias Jerboa.Format.Body.Attribute
+  alias Jerboa.Params
+  alias Fennec.TURN
+  @lifetime 10 * 60
+
+  @spec service(Params.t, map, TURN.t) :: {Params.t, TURN.t}
+  def service(x, changes, turn_state) do
+    request_status =
+      {:valid, x, %{}}
+      |> maybe(&verify_existing_allocation/4, [changes, turn_state])
+      |> maybe(&verify_requested_transport/2)
+      |> maybe(&verify_unknown_attributes/2)
+
+    case request_status do
+      {:error, error_code} ->
+        {%{x | attributes: [error_code]}, turn_state}
+      {:ok, {new_params, new_turn_state}} ->
+        {new_params, new_turn_state}
+      {:valid, _, state} ->
+        allocate(x, state, changes, turn_state)
```

Comment:
`state` is a state of message verification. All those functions that return `:valid`, also return the abstract state of this processing. This will be required as soon as there will be support for `DONT-FRAGMENT` attribute.